diff --git a/modules/juce_audio_plugin_client/VST3/juce_VST3_Wrapper.cpp b/modules/juce_audio_plugin_client/VST3/juce_VST3_Wrapper.cpp
index bb2ae6535..df37b6bf7 100644
--- a/modules/juce_audio_plugin_client/VST3/juce_VST3_Wrapper.cpp
+++ b/modules/juce_audio_plugin_client/VST3/juce_VST3_Wrapper.cpp
@@ -2116,6 +2116,16 @@ public:
         if (host != hostContext)
             host.loadFrom (hostContext);
 
+    	
+#ifdef JUCE_VST3_ENABLE_PASS_HOST_CONTEXT_TO_AUDIO_PROCESSOR_ON_INITIALIZE
+        getPluginInstance().handleVST3HostContext(
+           hostContext,
+           this->host.get(),
+           this->comPluginInstance.get(),
+           this->juceVST3EditController.get()
+        );
+#endif
+
         processContext.sampleRate = processSetup.sampleRate;
         preparePlugin (processSetup.sampleRate, (int) processSetup.maxSamplesPerBlock);
 
